addons/token-exchange/token_exchanger_agent.go

```diff
@@ -25,13 +25,17 @@ const (
 
 func NewAgentCommand() *cobra.Command {
 	o := NewAgentOptions()
-	cmd := controllercmd.
-		NewControllerCommandConfig(TokenExchangeName, version.Info{Major: "0", Minor: "1"}, o.RunAgent).
-		NewCommand()
+	cmdConfig := controllercmd.
+		NewControllerCommandConfig(TokenExchangeName, version.Info{Major: "0", Minor: "1"}, o.RunAgent)
+
+	cmd := cmdConfig.NewCommand()
 	cmd.Use = TokenExchangeName
 	cmd.Short = "Start the token exchange addon agent"
 
 	o.AddFlags(cmd)
+
+	flags := cmd.Flags()
+	flags.BoolVar(&cmdConfig.DisableLeaderElection, "disable-leader-election", true, "Disable leader election for the agent")
```

Comment:
Leader election does not impact performance. We should keep it to allow HA for addons.

Reply:
From what i tested, it actually did impact the performance of the agent. 
Not sure how the HA would be impacted since we only have a single replica running in the MC

Reply:
We can always scale it up. Leader election usually happens very fast. Might need different configuration but we should still try to keep it.

Reply:
I have tried to change it.  I have to investigate this issue deeper with teams